### PR TITLE
Adding iOS docs

### PR DESCRIPTION
**Pull request by victor-livefront**
Adding iOS documentation for the passwordless docs. Mostly from the iOS repo readme, but removed any references to the the repo itself, example project, and things covered by the overarching documentation.

### CODE REVIEW

**Review by abergs — CHANGES_REQUESTED** [review]
Pending review, just blocking merge until bits are available for devs to consume

File: src/guide/frontend/ios.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+---
+title: iOS Client Reference
+---
+
+# iOS Client Reference
+
+The purpose of this SDK is to be able to easily integrate [Passkeys](https://developer.apple.com/passkeys/) into your iOS app with the help of the [Passwordless.dev](https://bitwarden.com/products/passwordless/) management system.
```
Comment (jonashendrickx) [inline src/guide/frontend/ios.md:7]: We probably require an extra section mentioning requirements:
- Permissions & capabilities
- Minimum iOS version

Comment (victor-livefront) [reply src/guide/frontend/ios.md:7]: Ah good call out. I'll add minimum iOS version and capabilities.

Comment (victor-livefront) [reply src/guide/frontend/ios.md:7]: Updated.

Comment (abergs) [reply src/guide/frontend/ios.md:7]: @victor-livefront Related, de-capitalize 'passkeys'

File: src/guide/frontend/ios.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+let verifyToken = try await passwordlessClient.register(token: registrationToken)
+```
+
+3. Pass the verification token to your public RESTful API to verify and return an authorization token. The user is now logged in to your app.
+
+![Registration Flow](./ios/Registration.gif)
+
+### Sign in
+
+There are two main approaches for passkey sign in: auto fill and manual entry.
```
Comment (abergs) [inline src/guide/frontend/ios.md:59]: We should probably mention signInWithDiscoverable and signInWithUserId briefly, just to explain them, but we don't need to to guide the developer too much.

**Comment by abergs** [comment]
LGTM. @fschillingeriv A friendly ping that we'll be adding this iOS SDK documentation to the docs, in case you want to review or make adjustments.

**Review by jonashendrickx — APPROVED** [review]

**Review by jonashendrickx — DISMISSED** [review]
Was able to follow the instructions and set everything up.

**Review by abergs — APPROVED** [review]

**Review by abergs — APPROVED** [review]

**Pull request merged**